src/nbl/video/SOpenGLContextLocalCache.cpp

```diff
@@ -282,6 +284,14 @@ void SOpenGLContextLocalCache::flushStateGraphics(IOpenGL_FunctionTable* gl, uin
             }
 
             currentState.pipeline.graphics.pipeline = nextState.pipeline.graphics.pipeline;
+
+            if (currentState.pipeline.graphics.pipeline->getObjectDebugName().size())
+            {
+                const GLuint glname = currentState.pipeline.graphics.usedPipeline;
+                const auto& dbgnm = currentState.pipeline.graphics.pipeline->getObjectDebugName();
+
+                gl->extGlObjectLabel(GL_PROGRAM_PIPELINE, glname, dbgnm.size(), dbgnm.c_str());
+            }
```

Comment:
why is this here? inside of a flush state!?

Reply:
becasue thats where GL object is created